572. Subtree of Another Tree.py

```diff
@@ -0,0 +1,28 @@
+class Solution:
+    def isSubtree(self, s, t):
+        """
+        :type s: TreeNode
+        :type t: TreeNode
+        :rtype: bool
+        """
+        def find(s, t):
+            sames = False
+            
+            if not s or not t:
+                return False
+            if s.val == t.val:
+                sames = same(s,t)
+            
+            return sames or find(s.left, t) or find(s.right, t)
+            
+        def same(s, t):
+            if not s and not t:
+                return True
+            elif s and not t or not s and t:
```

Comment:
we can simplify this kind of. Look at your old solution.

The idea is we first check if `not s and not t` which means `s` and `t` are both `None`.
Then only if the above is not true, we check if one of them is `None` but the other isn't.
Since we have already verified that they both aren't `None` the only other cases are as follows:
1) One of `{s, t}` is`None`
2) `{s, t}` both not None.
We only want to do this branch in case 1, and your logic is indeed correct, but we can simplify it to:
`not s or not t` (and get rid of the final `False` since we check for `s.val == t.val`)